pkg/backup/actions/csi/pvc_action.go

```diff
@@ -570,3 +581,485 @@ func NewPvcBackupItemAction(f client.Factory) plugincommon.HandlerInitializer {
 		}, nil
 	}
 }
+
+func (p *pvcBackupItemAction) getVolumeSnapshotReference(
+	ctx context.Context,
+	pvc corev1api.PersistentVolumeClaim,
+	backup *velerov1api.Backup,
+) (*snapshotv1api.VolumeSnapshot, error) {
+	vgsLabelKey := backup.Spec.VolumeGroupSnapshotLabelKey
+	group, hasLabel := pvc.Labels[vgsLabelKey]
+
+	if vgsLabelKey != "" && hasLabel && group != "" {
```

Comment:
Let's add a wrap for all the return err branches, so that we have enough context during troubleshooting.